### ISSUE TITLE
[feature request] Powered elytra boost

### ISSUE THREAD

**Issue by MuteTiefling** [TeamPneumatic/pnc-repressurized#383]
I was thinking this could be a fun little interaction. Basically use the jump boost upgrade in the pneumatic boots to also boost your elytra flight. 

In my testing, it seems that the jump boost currently cannot be activated while in the air already, even while flying with elytra. So while the current implementation is great for getting airborne, it would be awesome if it could also help sustain flight.

**Comment by desht** [comment]
By "jump boost", do you mean the Jet Boots upgrade?

**Comment by MuteTiefling** [comment]
Yeah, the jet boots.

**Comment by ProfAggressor** [comment]
The Vortex Cannon already boosts Elytra, with some serious oomph.  A boost from the jet boots would be nice, though, and/or an Elytra upgrade for the Pneumatic Chestplate.  Currently, Elytra and Jet Boots don't really play well together.  The Jet Boots kind of want to be in charge, so they limit your speed and ability to glide, but the Elytra simultaneously make your movement very drifty and difficult to precisely control.

**Comment by MuteTiefling** [comment]
At this point, this is a pretty old suggestion and I made it when I knew a lot less about the mod. It honestly doesn't make a whole lot of sense anymore.

**Comment by desht** [comment]
It would be nice though for the Elytra and Jet Boots to get along a bit better (in particular, allowing gliding as normal with the Elytra when Jet Boots are active, but allow firing Jet Boots while gliding to get a speed boost).  So, disable the normal Jet Boots hover functionality if actively elytra-flying... might not be too hard, will need to experiment.

**Comment by tDegan** [comment]
What's the status on this? Especially with the new elytra upgrade, it'd be great if they would play nicely together.

Even just taking the current speed and direction when starting the jet boots and letting the jet boots control it from there instead of resetting to zero would be great.

**Comment by desht** [comment]
The Elytra (either as standalone Elytra or the Elytra Upgrade in the chestplate) and Jet Boots do play pretty nicely together.  You can glide as normal with the Elytra (double tap Space to engage Elytra as normal), then briefly fire the Jet Boots for a bit of speed and/or altitude.  When you stop firing the Jet Boots, Elytra gliding resumes.

Elytra takes durability damage as normal, Elytra Upgrade uses a very small amount of air (1 mL/t while gliding) instead. And Elytra upgrade is toggleable just like any other upgrade.

Have a play with it, see what you think.  Standalone Elytra support is in 1.16 and 1.18, Elytra Upgrade needs v3.3.0 for 1.18.2.

**Comment by tDegan** [comment]
My problem is this: When I glide with the elytra and I then activate the jet boots, my movement speed is set to zero. So instead of giving me a boost, they first slow me down before slowly accelerating me again.

The other direction works fine.

I am on 3.3.0-99

**Comment by desht** [comment]
Yeah, that's not ideal, I agree. I'll see if I can make that smoother.
